### PR TITLE
[thread-cert] fix case 7.1.6 intermittent failure

### LINKED ISSUE #5942: [github-actions] Cert_7_1_06_BorderRouterAsLeader.py fails intermittently
**Describe the bug**

tests/scripts/thread-cert/Cert_7_1_06_BorderRouterAsLeader.py

**Console/log output**

[logs_44948.zip](https://github.com/openthread/openthread/files/5680578/logs_44948.zip)
[packet-verification-pcaps.zip](https://github.com/openthread/openthread/files/5680589/packet-verification-pcaps.zip)

### PR DESCRIPTION
**Pull request by FiveDimensions**
Fixes #5942

### CODE REVIEW

File: tests/scripts/thread-cert/Cert_7_1_06_BorderRouterAsLeader.py [hunk 1]
```diff
@@ -245,33 +245,34 @@ def verify(self, pv):
...
+                filter_LLANMA().\
+                filter_mle_cmd(MLE_DATA_RESPONSE).\
+                filter(lambda p: {
+                                  NETWORK_DATA_TLV,
+                                  SOURCE_ADDRESS_TLV,
+                                  LEADER_DATA_TLV
+                                 } <= set(p.mle.tlv.type) and\
+                                 {
+                                  NWD_BORDER_ROUTER_TLV,
+                                  NWD_BORDER_ROUTER_TLV,
```
Comment (librasungirl) [inline tests/scripts/thread-cert/Cert_7_1_06_BorderRouterAsLeader.py:259]: meaningless to have the identical items in the `set`

**Review by librasungirl — APPROVED** [review]
Approved here to fix the failure
May have a separate PR to unify the BR related tests once #5881 is done.

File: tests/scripts/thread-cert/Cert_7_1_06_BorderRouterAsLeader.py (hunk 1 above)
Comment (FiveDimensions) [reply tests/scripts/thread-cert/Cert_7_1_06_BorderRouterAsLeader.py:259]: will file another PR

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**